_posts/2021-03-15-micrometer-prometheus-openshift.adoc

```diff
@@ -0,0 +1,207 @@
+
+```
+http_server_requests_seconds_count{method="GET",outcome="SUCCESS",status="200",uri="/hello",} 1.0
+http_server_requests_seconds_sum{method="GET",outcome="SUCCESS",status="200",uri="/hello",} 0.018198043
+```
+
+The same will happen if we're using the Kafka extension. All this configuration is well documented in https://quarkus.io/guides/micrometer[the Quarkus Micrometer guide].
+
+__How are these metrics consumed?__ As said, Micrometer exposes the metrics on the `/q/metrics` endpoint in order to be consumed by third parties like https://prometheus.io/[Prometheus]. OpenShift provides an embedded Prometheus instance that we can use. So, let's see what we need to do to integrate our Quarkus Applications metrics and the Prometheus embedded instance provided in OpenShift. 
+
+Basically, we need to complete the next following steps:
+
+1. Enable Monitoring for user-defined projects in OpenShift
+2. Deploy our Quarkus application
+3. Create a Service monitor in our project
+
+== Requirements
+
+- https://docs.openshift.com/container-platform/4.7/cli_reference/openshift_cli/getting-started-cli.html[Openshift CLI]
+- An Openshift instance
+
+== Step 1 - Enable Monitoring for user-defined projects in OpenShift
+
+In order to enable the Monitoring for user-defined projects, we need to follow the instructions https://docs.openshift.com/container-platform/4.7/monitoring/enabling-monitoring-for-user-defined-projects.html[from OpenShift documentation].
+
+In essence, we need to create a ConfigMap in the namespace `openshift-monitoring`.
+
+`cluster-monitoring-config.yaml`:
+[source, yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+    name: cluster-monitoring-config
+    namespace: openshift-monitoring
+data:
+    config.yaml: |
+    enableUserWorkload: true
+----
+
+And apply it:
+
+[source, bash]
+----
+oc apply -f cluster-monitoring-config.yaml
+----
+
+NOTE: For 4.5 or previous versions of OpenShift, you need to use instead:
+
+[source, yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+    name: cluster-monitoring-config
+    namespace: openshift-monitoring
+data:
+    config.yaml: |
+    techPreviewUserWorkload:
+        enabled: true
+----
+
+What OpenShift will do next, it's to automatically create the namespace `openshift-user-workload-monitoring` where will deploy some required resources for Prometheus.
+
+== 2. Deploy our Quarkus application
+
+Right now, the required infrastructure in OpenShift is ready, then let's continue with the Quarkus application. We'll implement a REST API and enable micrometer with prometheus, this is translated into having the next two Quarkus extensions into our `pom.xml` file:
```

Comment:
Instead of "Right now, ", what about "Once " or "When " or "After "

> When the required infrastructure in OpenShift is ready, we can continue with the Quarkus application. ...

Reply:
Fixed.